pkg/artifact/config.go

```diff
@@ -0,0 +1,208 @@
+		Name:   "github-app-id-secret",
+		Target: &cfg.GitHubAppIDSecret,
+		EnvVar: "GITHUB_APP_ID_SECRET",
+		Usage:  `The secret name & version containing the GitHub App ID.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-install-id",
+		Target: &cfg.GitHubAppID,
+		EnvVar: "GITHUB_INSTALL_ID",
+		Usage:  `The provisioned GitHub App installation ID.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-install-id-secret",
+		Target: &cfg.GitHubInstallIDSecret,
+		EnvVar: "GITHUB_INSTALL_ID_SECRET",
+		Usage:  `The secret name & version containing the GitHub App installation ID.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-private-key",
+		Target: &cfg.GitHubPrivateKey,
+		EnvVar: "GITHUB_PRIVATE_KEY",
+		Usage:  `The private key generated to call GitHub.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-private-key-secret",
+		Target: &cfg.GitHubPrivateKeySecret,
+		EnvVar: "GITHUB_PRIVATE_KEY_SECRET",
+		Usage:  `The secret name & version containing the GitHub App private key.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "bucket-name",
+		Target:  &cfg.BucketName,
+		EnvVar:  "BUCKET_NAME",
+		Usage:   `The name of the bucket that holds artifact logs files from GitHub`,
+		Example: "retry-lock-xxxx",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "events-table-id",
+		Target: &cfg.EventsTableID,
+		EnvVar: "EVENTS_TABLE_ID",
+		Usage:  `The events table ID within the dataset.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "artifacts-table-id",
+		Target: &cfg.ArtifactsTableID,
+		EnvVar: "ARTIFACTS_TABLE_ID",
+		Usage:  `The artifacts table ID within the dataset.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "project-id",
+		Target: &cfg.ProjectID,
+		EnvVar: "PROJECT_ID",
+		Usage:  `Google Cloud project ID.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "dataset-id",
+		Target: &cfg.DatasetID,
+		EnvVar: "DATASET_ID",
+		Usage:  `BigQuery dataset ID.`,
+	})
+
+	f.IntVar(&cli.IntVar{
+		Name:    "batch-size",
+		Target:  &cfg.BatchSize,
+		EnvVar:  "BATCH_SIZE",
+		Default: 100,
+		Usage:   `The number of items to process in this execution`,
+	})
+
+	return set
+}
+
+func (cfg *Config) ReplaceSecrets(ctx context.Context) error {
+	// load any secrets from secret manager
+	sm, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create secret manager client: %w", err)
+	}
+	defer sm.Close()
+
+	cfg.GitHubAppID, err = readSecretText(ctx, sm, cfg.GitHubAppIDSecret, cfg.GitHubAppID)
+	if err != nil {
+		return fmt.Errorf("failed to process app id secret: %w", err)
+	}
+	cfg.GitHubInstallID, err = readSecretText(ctx, sm, cfg.GitHubInstallIDSecret, cfg.GitHubInstallID)
+	if err != nil {
+		return fmt.Errorf("failed to process install id secret: %w", err)
+	}
+	cfg.GitHubPrivateKey, err = readSecretText(ctx, sm, cfg.GitHubPrivateKeySecret, cfg.GitHubPrivateKey)
+	if err != nil {
+		return fmt.Errorf("failed to process private key secret: %w", err)
+	}
+	return nil
+}
```

Comment:
I feel like this is a work around for dealing with Beam, can we just use the native integration with cloud run jobs and secret manager to mount these instead?

Reply:
Ugh. Yes. I can do that. The deployment side of this is already a mess so throwing more chaos into it isn't going to make it that much worse.

Reply:
Yea, I think it'd be nicer if we could load these via envvars from Cloud Run natively.

Reply:
That _should_ also drop the runtime dep on secretmanager, which will be nice for the dep tree.

Reply:
Yeah, I'm working on this one.